### ISSUE TITLE
Content suggestion: Remove <length-percentage> and similar pages

### ISSUE THREAD

**Issue by foolip** [mdn/content#13066]
## What is the new suggestion?

Remove https://developer.mozilla.org/en-US/docs/Web/CSS/length-percentage and similar pages.

## Why is it important or useful?

https://github.com/mdn/browser-compat-data/pull/14971 is a PR to remove it in BCD, because support data is just the union of two other features. If we remove it from BCD, either the MDN page should be removed, or it includes the two compat tables for &lt;length> and &lt;percentage>.

## Other supporting information

My weak opinion is that exposing these spec abstractions to readers is a bit weird, but that argument also applies to &lt;length> and &lt;percentage>. Perhaps they're useful to avoid spelling out all the allowed values for each property.

**Comment by foolip** [comment]
> > that argument also applies to  and . Perhaps they're useful to avoid spelling out all the allowed values for each property.
> 
> Yes, [`<length>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length) has value that [`<length-percentage>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length-percentage) doesn't. All `<length-percentage>` can say really is that it "represents a value that can be either a <length> or a <percentage>" and we can just as easily, and probably more usefully, say that in the places we want to refer to it. But `<length>` can and does say a lot about allowed units and their meanings, and we really don't want to duplicate all that.
> 
> So as a pragmatic matter I think deleting `<length-percentage>` would be OK and deleting `<length>` would not.

That sounds good to me.

> More generally, BCD for data types is a bit weird. You can't do anything with a data type on its own, so what does "support" mean? Take for instance [`<flex>`](https://developer.mozilla.org/en-US/docs/Web/CSS/flex_value). We say it's supported since Chrome 29. The page refers to its use in [`grid-template-rows`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-rows). But you can't use `<flex>` there in Chrome 29 of course, because you can't use `grid-template-rows` in Chrome 29. So what does it mean, to a developer, to say that a CSS data type is supported?
> 
> So sometimes I feel like CSS data types should not have BCD at all.

Sometimes, perhaps. But do you agree that [`<length>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length#browser_compatibility) is important to keep since it's the place that holds support for the `ic` unit? Is the distinction that `<length>` has subfeatures, but `<flex>` does not?

**Comment by wbamberg** [comment]
> > More generally, BCD for data types is a bit weird. You can't do anything with a data type on its own, so what does "support" mean? Take for instance [`<flex>`](https://developer.mozilla.org/en-US/docs/Web/CSS/flex_value). We say it's supported since Chrome 29. The page refers to its use in [`grid-template-rows`](https://developer.mozilla.org/en-US/docs/Web/CSS/grid-template-rows). But you can't use `<flex>` there in Chrome 29 of course, because you can't use `grid-template-rows` in Chrome 29. So what does it mean, to a developer, to say that a CSS data type is supported?
> > So sometimes I feel like CSS data types should not have BCD at all.
> 
> Sometimes, perhaps. But do you agree that [`<length>`](https://developer.mozilla.org/en-US/docs/Web/CSS/length#browser_compatibility) is important to keep since it's the place that holds support for the `ic` unit? Is the distinction that `<length>` has subfeatures, but `<flex>` does not?

Well... this is outside the scope of your original issue, which I agree with. And I don't think we will resolve this wider issue now, and we have bigger things to deal with.

But it's not really about subfeatures. When we provide BCD we are saying to a developer "you can use this thing". But you can't use a data type on its own, it's just a value for properties, which are the things you use. So it doesn't really mean anything. 

Consider https://developer.mozilla.org/en-US/docs/Web/CSS/basic-shape#browser_compatibility, the compat for `<basic-shape>`. It has subfeatures, and lists some browser versions. It's a data type that can be used, the page says, as a value for the [`clip-path`](https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path), [`shape-outside`](https://developer.mozilla.org/en-US/docs/Web/CSS/shape-outside), and [`offset-path`](https://developer.mozilla.org/en-US/docs/Web/CSS/offset-path) properties. But all these properties have different BCD stories - and have to document themselves when particular values of `<basic-shape>` can be used. So how could the BCD for `<basic-shape>` possibly tell developers anything useful about when they can use the value?

It's rather like the way we do/do not provide BCD for dictionaries. On its own a dictionary doesn't have any BCD, what does is the particular set of values an API could accept.

I don't have a good solution to this really, and there are types like `<length>` which are so widely used that it would be very awkward to duplicate the story for every property that uses it. So perhaps what we do now is the best pragmatic option. But I just had to mention it.

**Comment by foolip** [comment]
I agree that the `<bla>` entries don't mean anything by themselves and it's a bug that we have to fill in data for it in BCD. I guess that's out of scope of this issue though, the `<length>` and `<percentage>` entries will remain.

At this point, are PRs to remove BCD tables from pages like https://developer.mozilla.org/en-US/docs/Web/CSS/length-percentage welcome?

**Comment by teoli2003** [comment]
We discussed this today and we think there are a lot of open points to discuss about it:
- How does this affect links from css formal sections-
- Should we keep these content pages without bcd info on them?
- @estelle. More than the <xxx-percentage> pages, how do we document types nowadays: not all are documented in the same way; what are the categories?

Let's convert this into a discussion.
